### PR TITLE
cmd/geth: added --mainnet flag

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
closes https://github.com/ethereum/go-ethereum/issues/21795

TODO:
* check that genesis is okay

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
@holiman I remembered in the discussion you said something about checking that the genesis is okay.
But I don't know how to do it, any hints?

**Comment by holiman** [comment]
Sorry, I missed that you had asked me something. So, I'm not 100% sure you need to do anything in particular. But check that "odd stuff" doesn't happen e.g when you do 
```
geth --goerli init
geth --mainnet 
```
or 
```
geth --goerli
ctrl-c
geth --mainnet --datadir=~/.ethereum/geth/goerli
```
Also, maybe we should check the result of doing:
```
geth --mainnet --networkid=1337
```
not sure what I would prefer, probably that the `networkid=1337` "wins", but maybe it's better to exit with an error.

**Comment by MariusVanDerWijden** [comment]
So `mainnet -> rinkeby` produces an error. However init with `rinkeby` then `mainnet` does not error (which it should)

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -1803,7 +1813,7 @@ func MakeChainDatabase(ctx *cli.Context, stack *node.Node) ethdb.Database {
 }
 
 func MakeGenesis(ctx *cli.Context) *core.Genesis {
-	var genesis *core.Genesis
+	genesis := core.DefaultGenesisBlock()
```
Comment (holiman) [inline cmd/utils/flags.go:1816]: What's the reason for this change? It looks like it previously might have returned `nil` in some cases, and probably error on the caller-side, whereas now it will default to mainnet in those cases. 

Is this a 'safe' change to do?

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -131,6 +131,10 @@ var (
 		Usage: "Explicitly set network id (integer)(For testnets: use --ropsten, --rinkeby, --goerli instead)",
 		Value: eth.DefaultConfig.NetworkId,
 	}
+	MainnetFlag = cli.BoolFlag{
+		Name:  "mainnet",
+		Usage: "Ethereum network: main network of the ETH asset",
```
Comment (holiman) [inline cmd/utils/flags.go:136]: Maybe just
```suggestion
		Usage: "Ethereum mainnet",
```
At least drop that "of the ETH asset" :)

File: cmd/utils/flags.go (hunk 1 above)
Comment (MariusVanDerWijden) [reply cmd/utils/flags.go:1816]: No, that's an artifact of testing

Comment (MariusVanDerWijden) [reply cmd/utils/flags.go:1816]: Thanks for noticing!

**Review by holiman — APPROVED** [review]
LGTM
